Suppress CA1707 by default in test projects

## Background and motivation

Test projects usually have long, underscore-separated method names, like `AppendXmlDocName_WithNullMember_ThrowsArgumentNullException` or `AppendXmlDocName_WithField_ReturnsCorrectValue`, that trigger [code quality warning CA1707](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1707).

While said warning has an agreeable rationale, I find myself routinely disabling CA1707 on test projects, like this. This is exactly the kind of repetitive task I wanted to avoid with Buildvana SDK.

## Proposed enhancement

Disable CA1707 by default in test projects.

## Implementation proposals

- Add a public-facing property `AllowUnderscoresInMemberNames`.
- If said property is left unset by project / common files, set it to `true` for test projects, `false` for other projects.
- If the property is `true`, suppress warning CA1707 by appending the string `";CA1707"` to the `NoWarn` property.

## Usage examples

Normal usage: nothing to do in projects / common files. (The Buildvana way! 😄)

Enable CA1707 in a test project:

```XML
  <AllowUnderscoresInMemberNames>false</AllowUnderscoresInMemberNames>
```

## Risks

If a test project already suppresses CA1707, the warning will appear twice in the `NoWarn` property: no consequences AFAIK.

--- a/src/Buildvana.Sdk/Modules/StandardAnalyzers/BeforeModules.targets
+++ b/src/Buildvana.Sdk/Modules/StandardAnalyzers/BeforeModules.targets
@@ -18,4 +18,13 @@
     <UsePublicApiAnalyzers>false</UsePublicApiAnalyzers>
   </PropertyGroup>
 
+  <!-- AllowUnderscoresInMemberNames default = true only on test projects. -->
+  <PropertyGroup Condition="'$(AllowUnderscoresInMemberNames)' == ''">
+    <AllowUnderscoresInMemberNames>false</AllowUnderscoresInMemberNames>
+    <AllowUnderscoresInMemberNames Condition="$(BV_IsTestProject)">true</AllowUnderscoresInMemberNames>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(AllowUnderscoresInMemberNames)' != 'true'">
+    <AllowUnderscoresInMemberNames>false</AllowUnderscoresInMemberNames>
+  </PropertyGroup>
+
 </Project>

--- a/src/Buildvana.Sdk/Modules/StandardAnalyzers/Module.targets
+++ b/src/Buildvana.Sdk/Modules/StandardAnalyzers/Module.targets
@@ -48,4 +48,10 @@
     <AdditionalFiles Include="$(MSBuildProjectDirectory)\PublicAPI.Unshipped.txt" />
   </ItemGroup>
 
+  <!-- Suppress warning CA1707 if AllowUnderscoresInMemberNames is true. -->
+  <PropertyGroup Condition="'$(AllowUnderscoresInMemberNames)' == 'true'">
+    <NoWarn Condition="'$(NoWarn)' != ''">$(NoWarn);</NoWarn>
+    <NoWarn>$(NoWarn)CA1707</NoWarn>
+  </PropertyGroup>
+
 </Project>

